ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activitySection.component.html

```diff
@@ -0,0 +1,79 @@
+        <div *ngIf="isConditional(block)">
+            <ddp-conditional-block [block]="block"
+                                   [readonly]="readonly"
+                                   [validationRequested]="validationRequested"
+                                   [studyGuid]="studyGuid"
+                                   [activityGuid]="activityGuid">
+            </ddp-conditional-block>
+        </div>
+        <div *ngIf="isQuestion(block)">
+            <ddp-activity-question [block]="block"
+                                   [readonly]="readonly"
+                                   [validationRequested]="validationRequested"
+                                   [studyGuid]="studyGuid"
+                                   [activityGuid]="activityGuid"
+                                   (componentBusy)="componentBusy.emit($event)">
+            </ddp-activity-question>
+        </div>
+        <div *ngIf="isInstitution(block)">
+            <ddp-institutions-form [block]="block"
+                                   [studyGuid]="studyGuid"
+                                   [readonly]="readonly"
+                                   [validationRequested]="validationRequested"
+                                   (validStatusChanged)="updateValidationStatusInSection(block.id, $event)"
+                                   (componentBusy)="componentBusy.emit($event)">
+            </ddp-institutions-form>
+        </div>
+        <div *ngIf="isMailAddress(block)">
+            <ddp-address-embedded [block]="block"
+                                  [country]="config.supportedCountry"
+                                  [readonly]="readonly"
+                                  [activityGuid]="activityGuid"
+                                  (validStatusChanged)="updateValidationStatusInSection(block.id, $event)"
+                                  [validationRequested]="validationRequested"
+                                  (componentBusy)="componentBusy.emit($event)">
+            </ddp-address-embedded>
+        </div>
+    </ng-template>
+
+    <div *ngIf="isContent(block)">
+        <ddp-activity-content [block]="block"></ddp-activity-content>
+    </div>
+    <div *ngIf="isGroup(block)">
+        <ddp-group-block [block]="block"
+                         [readonly]="readonly"
+                         [validationRequested]="validationRequested"
+                         [studyGuid]="studyGuid"
+                         [activityGuid]="activityGuid">
+        </ddp-group-block>
+    </div>
+    <div *ngIf="isActivityBlock(block)">
+        <ddp-activity-block [block]="block"
+                            [readonly]="readonly"
+                            [validationRequested]="validationRequested"
+                            [studyGuid]="studyGuid"
+                            [parentActivityInstanceGuid]="activityGuid"
+                            (validStatusChanged)="updateValidationStatusInSection($event.id, $event.value)"
```

Comment:
can it be block.id instead of $event.id?

Reply:
No.
That activityBlock has a list of embedded activities (childInstances) into itself with different `instanceId`.
They can emit their own `validationStatus`.

Reply:
but we didn't care about it before and considered only one returned result for the whole block. Was it a wrong approach?

Reply:
I feel like it was an incorrect decision. 
We missed the fact that `ActivityBlock` can contain several `embeddedActivities` (like 'Siblings'/ 'Children').
Previously we used one a key for `activityBlock`. However a value (validation status) was overriden from one `embeddedActivity` validation status to another.